data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/beacon/ValidatorResponse.java

```diff
@@ -59,15 +62,20 @@ public ValidatorResponse(
     this.validator = validator;
   }
 
-  public static ValidatorResponse fromState(final BeaconState state, final Integer index) {
+  public static Optional<ValidatorResponse> fromState(
+      final BeaconState state, final Integer index) {
+    if (index >= state.getValidators().size()) {
+      throw new BadRequestException("Validator index out of bounds: " + index);
+    }
```

Comment:
If this is going to throw an exception if the index is too big then it doesn't need to return `Optional` because it will never return empty.  I wouldn't have expected a validator index being too big to be a bad request, it's just not found and so the validator wouldn't be included in the results.

Reply:
was getting index out of bounds, which you didnt want to catch, so this was option b, i'll have to have another look.

Reply:
Just return `Optional.empty()` here instead of throwing an exception.